app/src/main/java/com/example/android/sunshine/app/DetailFragment.java

```diff
@@ -152,23 +157,22 @@ public void onActivityCreated(Bundle savedInstanceState) {
 
     void onLocationChanged(String newLocation) {
         // replace the uri, since the location has changed
-        Uri uri = mUri;
+        Uri uri = this.uri;
         if (null != uri) {
             long date = WeatherContract.WeatherEntry.getDateFromUri(uri);
-            Uri updatedUri = WeatherContract.WeatherEntry.buildWeatherLocationWithDate(newLocation, date);
-            mUri = updatedUri;
+            this.uri = WeatherEntry.buildWeatherLocationWithDate(newLocation, date);
             getLoaderManager().restartLoader(DETAIL_LOADER, null, this);
         }
     }
 
     @Override
     public Loader<Cursor> onCreateLoader(int id, Bundle args) {
```

Comment:
Yeah, legacy stuff... 
While you make good points with this and also about splitting the `Utility` class, I think that these would be outside of the scope of this PR. But I took note of them, and I will bring these improvements in a future PR.